### PR TITLE
aws ipam added

### PR DESCRIPTION
**Pull request by umairgha**

### CODE REVIEW

File: aws-ipam.tf [hunk 1]
```diff
@@ -0,0 +1,46 @@
+# Create a service linked role
+resource "aws_iam_service_linked_role" "dataworks_aws_ipam_service_linked_role" { # only required for single account IPAM.
```
Comment (rbryett) [inline aws-ipam.tf:2]: Is this used by our users and is there an accompanying PR?

File: aws-ipam.tf [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+  ipam_scope_id  = aws_vpc_ipam.dataworks_aws_ipam.private_default_scope_id
+}
+
+# Provision IPAM Pool CIDR for top-level pool
+resource "aws_vpc_ipam_pool_cidr" "dataworks_aws_ipam_global_pool_cidr" {
+  ipam_pool_id = aws_vpc_ipam_pool.dataworks_aws_ipam_global_pool.id
+  cidr         = local.global_pool_cidr
+}
+
+# Rreate regional pools
```
Comment (rbryett) [inline aws-ipam.tf]: Small typo - Rreate > Create

File: aws-ipam.tf (hunk 1 above)
Comment (umairgha) [reply aws-ipam.tf:2]: No this service linked role is only for IPAM to manage the resources

File: aws-ipam.tf (hunk 2 above)
Comment (umairgha) [reply aws-ipam.tf]: Corrected

File: variables.tf [hunk 3]
```diff
@@ -8,3 +8,14 @@ variable "region" {
   type    = string
   default = "eu-west-2"
 }
+
```
Comment (rbryett) [inline variables.tf]: Remove these if not required rather than leave as comments

File: .terraform.lock.hcl [hunk 4]
```diff
@@ -0,0 +1,21 @@
+# This file is maintained automatically by "terraform init".
+# Manual edits may be lost in future updates.
+
```
Comment (rbryett) [inline .terraform.lock.hcl:3]: Can you put a note on https://projects.ucd.gpn.gov.uk/browse/DW-8217 that this .terraform.lock.hcl has already been committed as part of this IPAM work please.
Just makes it easier for when people come to pick up the ticket.

File: variables.tf (hunk 3 above)
Comment (umairgha) [reply variables.tf]: done

**Review by rbryett — APPROVED** [review]

File: .terraform.lock.hcl (hunk 4 above)
Comment (umairgha) [reply .terraform.lock.hcl:3]: OK

**Pull request merged**